### DISCUSSION TITLE
How to test the child component Controller input when I pass useForm as a prop

### DISCUSSION THREAD

**Discussion by slamdunkYYY** [react-hook-form/react-hook-form#9112] [Q&A]
I have a children component as a shared form using with different parent components, those parent components have different buttons and their actions.

My parent component is like:
```
...
const form = useForm({
    defaultValues: {
        name: '',
    }
})

const { handleSubmit } = form

const onSubmit = (data) => { ... }

return (
    <>
        <Child form={form} />
        <button onClick={handleSubmit(onsubmit)} />
    </>
)
...
```

The child component is:
```
...
const {register, control } = props.form

return (
    <form>
        <Controller control={control} name="name" render={(field, fieldState) => <input .../>} />
    </form>
)
...
```

How can I test the children component input behavior?

I used a `jest.mock` to `react-hook-form`:
```
jest.mock('react-hook-form', () => {
    ...jest.requireActural('react-hook-form')
    Controller: ()=><></>
    useForm:()=>({
        Control: ()=>({}),
        handleSubmit: ()=>jest.fn()
    })
})
```
But this mock cannot get the input on the DOM. On the `screen.debug`, I can only see a `<form>` tag renders, but not the content.
Here is the code sandbox link: https://codesandbox.io/s/react-hook-form-v7-controller-forked-fwjqxg?file=/src/index.js

**Comment by bluebill1049** [comment]
you can refer to those tests example in this folder: https://github.com/react-hook-form/react-hook-form/tree/master/src/__tests__
